packages/mls-client/src/AsyncStream.ts

```diff
@@ -0,0 +1,77 @@
+type Value<T, V> = V extends undefined ? T : V
+
+type ResolveValue<T, V> = {
+  value: Value<T, V> | undefined
+  done: boolean
+}
+
+type ResolveNext<T, V> = (resolveValue: ResolveValue<T, V>) => void
+
+type TransformValue<T, V> = (value: T) => Value<T, V>
+
+export class AsyncStream<T, V = undefined> {
+  #done = false
+  #resolveNext: ResolveNext<T, V> | null
+  #queue: Value<T, V>[]
+  #transformValue?: TransformValue<T, V>
+
+  stopCallback: (() => void) | undefined = undefined
+
+  constructor(
+    transformValue: V extends undefined ? undefined : TransformValue<T, V>
+  ) {
+    this.#queue = []
```

Comment:
Hash prefixes are what all the cool kids in javascript are doing now?

Reply:
haha. it's updated syntax for [private class properties](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Classes/Private_properties).

Reply:
`private` is a TypeScript only modifier. this new syntax is part of the JavaScript language.

Reply:
Ahhh. Yet another reminder that Typescript only lives in our hearts and minds, but definitely not the JS engine.